pyvrp/cpp/search/Route.cpp

```diff
@@ -80,55 +121,52 @@ bool Route::overlapsWith(Route const &other, int const tolerance) const
 
 void Route::update()
 {
-    auto const oldNodes = nodes;
-    setupNodes();
+    nodes.clear();
+    auto *node = n(&startDepot);
 
     Load load = 0;
     Distance distance = 0;
     Distance reverseDistance = 0;
-    bool foundChange = false;
 
-    for (size_t pos = 0; pos != nodes.size(); ++pos)
+    while (!node->isDepot())
     {
-        auto *node = nodes[pos];
-
-        if (!foundChange && (pos >= oldNodes.size() || node != oldNodes[pos]))
-        {
-            foundChange = true;
-
-            if (pos > 0)  // change at pos, so everything before is the same
-            {             // and we can re-use cumulative calculations
-                load = nodes[pos - 1]->cumulatedLoad;
-                distance = nodes[pos - 1]->cumulatedDistance;
-                reverseDistance = nodes[pos - 1]->cumulatedReversalDistance;
-            }
-        }
-
-        if (!foundChange)
-            continue;
+        size_t const position = nodes.size();
+        nodes.push_back(node);
 
         load += data.client(node->client).demand;
         distance += data.dist(p(node)->client, node->client);
 
         reverseDistance += data.dist(node->client, p(node)->client);
         reverseDistance -= data.dist(p(node)->client, node->client);
 
-        node->position = pos + 1;
+        node->position = position + 1;
         node->cumulatedLoad = load;
         node->cumulatedDistance = distance;
         node->cumulatedReversalDistance = reverseDistance;
         node->twBefore
             = TWS::merge(data.durationMatrix(), p(node)->twBefore, node->tw);
+
+        node = n(node);
     }
 
+    load += data.client(endDepot.client).demand;
+    distance += data.dist(p(&endDepot)->client, endDepot.client);
+
+    reverseDistance += data.dist(endDepot.client, p(&endDepot)->client);
+    reverseDistance -= data.dist(p(&endDepot)->client, endDepot.client);
```

Comment:
Could we rename `reverseDistance`? `reverseDistance` implies that it's the distance from `endDepot` to `startDepot`. 

I think `reversalDistance` is fine since it's consistent with `cumulatedReversalDistance`, but it'd be even better to add "delta" somewhere. But then the name will get pretty verbose so I'm not really sure if that works.

Reply:
It's only used in `TwoOpt::evalWithinRoute`. I want to test whether that move actually adds any value, or can be removed. In the latter case this whole thing can go. The in-route 2OPT move evaluates reversing a route segment, which doesn't seem all that useful for VRPTW. But I'll let benchmarks for CVRP and VRPTW decide.

Reply:
I'd not mind replacing our `TwoOpt` with a generic, templated $k$-OPT, and then provide several such operators. That seems more useful than what we have currently with 2OPT.

Reply:
#285.